### DISCUSSION TITLE
Dealing with bleaching

### DISCUSSION THREAD

**Discussion by zhouyi0812** [flatironinstitute/CaImAn#1255] [Q&A]
Hello!
My movies have different degree of bleaching and I check there are different ways of dealing with it.
I am wondering how do you usually deal with it? I saw there is a debleach(self) function on the core function of caiman, if that is enough?
Thank you!

**Comment by EricThomson** [comment]
Once you extract a trace, applying `detrend_df_f()` will remove artifacts on long time scales like bleaching. That `debleach()` function is an interesting discovery: I have never seen it before. 😮  The movie object used to be a catch-all for many different methods, and this seems to be a vestige of those days. I'm not sure it should even still be in the code base 😄

**Comment by zhouyi0812** [comment]
Thank you! 
Just wonder if the right order would be: Getting denoised trace: estimates.C --> remove bleaching: detrend_df_f --> normalize trace to get df/f by z score

**Comment by EricThomson** [comment]
That would be a reasonable strategy.

**Comment by zhouyi0812** [comment]
Hello!
May I ask a follow up question?
How to know if detrend_df_f is using the denoised data? 
The demo is directly use cnm.estimates.detrend_df_f(quantileMin=8, 
                                      frames_window=540,
                                      use_residuals=False),
I was trying to cnm.estimates.detrend_df_f(denoised_trace[quantileMin=8, 
                                      frames_window=540,
                                      use_residuals=False]) but this is not necessary. 
Thank you!

**Reply by EricThomson** [reply]
The algorithm will automatically use denoised data if `use_residuals` is `False` (for 1p data it will just detrend, not normalize).

**Reply by zhouyi0812** [reply]
Hello! I wonder after I apply detrend_df_f on data, the output is caiman.source_extraction.cnmf.cnmf.CNMF. 
And if I want to get traces, can I go with this?
Thank you!!
good_inds = cnmfe_model.estimates.idx_components
denoised_traces = cnmfe_model.estimates.C[good_inds,:]
